### PR TITLE
Refactor constants, utils and main

### LINKED ISSUE #761: refactor Constants.h
`Constants.h` contains `#define` values for a whole host of different unrelated things. The problem with this approach is that changing anything in this file leads to huge recompiles, because loads of files include `Constants.h` (directly or indirectly). We should rather separate out these values into the appropriate headers (`Logger.h`, `Frame.h` etc)

### PR DESCRIPTION
**Pull request by confluence**
The main focus of this PR is to fix #761 and #744. It splits `Constants.h` and `Util.*` into multiple smaller files (currently in a `Util` subdirectory), and moves the messaging code from `Main.cc` into a new `SessionManager` class. Some gRPC-specific code has also been moved out of main, and some other common code has been rearranged.

Is this a sensible split? Should `Util` be called something else, like `Common`?

I have a few other refactoring suggestions, which could be done in this PR or in separate PRs:

* Move `MimeTypes` to `Util`
* Fix all the unnecessary relative imports (imports relative to the top-level directory work fine)
* Remove `using namespace std` everywhere -- we don't use it consistently anyway, and I think that it makes the code more confusing to read
* Fix the inconsistent use of namespaces
* Move `ProgramSettings` into the root directory -- it makes no sense for it to be in `SessionManager`, because the settings are not per-session
* Move `SimpleHttpServer` into the `SessionManager` directory?
* Move `Logger` into `Util`?

### CODE REVIEW

**Comment by pford** [comment]
I'm not sure I understand how refactoring 3 files into 15 Util/ files is a simplification of the code, especially when the lines are blurred e.g. file code in Casacore.h or File.h?  But I agree with putting constants where they are used when they are not shared.

**Comment by confluence** [comment]
> I'm not sure I understand how refactoring 3 files into 15 Util/ files is a simplification of the code, especially when the lines are blurred e.g. file code in Casacore.h or File.h? But I agree with putting constants where they are used when they are not shared.

The main motivation is to avoid putting all shared imports in a single header, so that any change to that header requires recompiling everything. Division into smaller and more specific files also makes the individual dependencies clearer conceptually.

I'm not completely sold on the Casacore subdivision; I could move the two formatting functions into a header specific to moments and divide the rest between File and Image. Does that sound like a better split?

**Comment by pford** [comment]
If it helps with the goal of reducing compilation, keep the Casacore header separate.

**Review by pford — APPROVED** [review]

File: src/GrpcServer/CartaGrpcService.cc [hunk 1]
```diff
@@ -88,3 +89,41 @@ grpc::Status CartaGrpcService::CallAction(
 
     return status;
 }
+
+GrpcManager::GrpcManager() : _selected_port(-1) {}
```
Comment (veggiesaurus) [inline src/GrpcServer/CartaGrpcService.cc:93]: is this the "dummy" constructor for tests?

**Review by veggiesaurus — APPROVED** [review]
looks fine. We can merge as soon as the conflicts are resolved

File: src/GrpcServer/CartaGrpcService.cc (hunk 1 above)
Comment (confluence) [reply src/GrpcServer/CartaGrpcService.cc:93]: It's the dummy constructor for the default case of running the backend with gRPC disabled.

**Pull request merged**